File: app/src/main/java/com/vkpapps/thunder/ui/fragments/GenericFragment.java
package com.vkpapps.thunder.ui.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.tabs.TabLayout;
import com.vkpapps.thunder.R;
import com.vkpapps.thunder.ui.fragments.viewpager.MyPagerAdapter;
import com.vkpapps.thunder.utils.PermissionUtils;

/***
 * @author VIJAY PATIDAR
 */
public class GenericFragment extends Fragment {
    public static final String PARAM_DESTINATION = "DESTINATION";
    private int destination = -1;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Bundle arguments = getArguments();
        if (arguments != null) {
            destination = arguments.getInt(PARAM_DESTINATION);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_generic, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        if (PermissionUtils.checkStoragePermission(view.getContext())) {
            TabLayout tabLayout = view.findViewById(R.id.tabLayout);
            ViewPager viewPager = view.findViewById(R.id.viewPager);
            tabLayout.setupWithViewPager(viewPager);

            MyPagerAdapter adapter = new MyPagerAdapter(getChildFragmentManager());
            PhotoFragment photoFragment = new PhotoFragment();
            adapter.addFragment(photoFragment, "Photos");
            AudioFragment audioFragment = new AudioFragment();
            adapter.addFragment(audioFragment, "Music");
            VideoFragment videoFragment = new VideoFragment();
            adapter.addFragment(videoFragment, "Videos");
            viewPager.setAdapter(adapter);

            if (destination > 0) {
                viewPager.setCurrentItem(destination);
            }
        } else {
            Navigation.findNavController(view).popBackStack();
            PermissionUtils.askStoragePermission(getActivity(), 101);
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();

    }
}

File: app/src/main/java/com/vkpapps/thunder/ui/fragments/GenericFragment.kt
package com.vkpapps.thunder.ui.fragments

import android.os.Bundle
import android.view.*
import androidx.cardview.widget.CardView
import androidx.fragment.app.Fragment
import androidx.navigation.NavController
import androidx.navigation.NavDirections
import androidx.navigation.Navigation
import androidx.viewpager.widget.ViewPager
import com.google.android.material.tabs.TabLayout
import com.vkpapps.thunder.R
import com.vkpapps.thunder.ui.activity.MainActivity
import com.vkpapps.thunder.ui.fragments.viewpager.MyPagerAdapter
import com.vkpapps.thunder.utils.PermissionUtils.askStoragePermission
import com.vkpapps.thunder.utils.PermissionUtils.checkStoragePermission

/***
 * @author VIJAY PATIDAR
 */
class GenericFragment : Fragment() {
    private var destination = 0
    private var navController: NavController? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val arguments = arguments
        if (arguments != null) {
            destination = arguments.getInt(PARAM_DESTINATION)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_generic, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setHasOptionsMenu(true)

        navController = Navigation.findNavController(view)
        if (checkStoragePermission(view.context)) {
            val tabLayout: TabLayout = view.findViewById(R.id.tabLayout)
            val viewPager: ViewPager = view.findViewById(R.id.viewPager)
            tabLayout.setupWithViewPager(viewPager)
            val adapter = MyPagerAdapter(childFragmentManager)
            val photoFragment = PhotoFragment()
            adapter.addFragment(photoFragment, "Photos")
            val audioFragment = AudioFragment()
            adapter.addFragment(audioFragment, "Music")
            val videoFragment = VideoFragment()
            adapter.addFragment(videoFragment, "Videos")
            viewPager.adapter = adapter
            viewPager.currentItem = destination
        } else {
            Navigation.findNavController(view).popBackStack()
            askStoragePermission(activity, MainActivity.ASK_PERMISSION_FROM_GENERIC_FRAGMENT)
        }
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        super.onCreateOptionsMenu(menu, inflater)
        val findItem = menu.findItem(R.id.menu_transferring)
        findItem?.actionView?.findViewById<CardView>(R.id.transferringActionView)?.setOnClickListener {
            navController?.navigate(object : NavDirections {
                override fun getArguments(): Bundle {
                    return Bundle()
                }

                override fun getActionId(): Int {
                    return R.id.action_navigation_files_to_transferringFragment
                }

            })
        }

    }

    companion object {
        const val PARAM_DESTINATION = "DESTINATION"
    }
}